### PR TITLE
Let's go ahead and merge testapp and the api2 preview now

### PR DESCRIPTION
**Pull request by afti-githobo**
## Description

This pull request adds an (actually working!) Android/iOS* port of testapp, plus an early revision of the new, higher-level Dart API.

Note that the latter sounds big, but it's not actually a breaking change for anything! The high-level API wraps the older, lower-level one; anything that was using the low-level API before will just continue to use it until someone decides to change it.

*_I don't have an iOS environment to work with, don't actually assume it works, someone should probably check it out_

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]

**Comment by kjawadDeveloper2** [comment]
@afti-githobo  a smaller PR  will be much better option. 

1) A PR for adding a new app. 
2) Another PR for the dart sdk changes. 


and so on.

**Comment by kjawadDeveloper2** [comment]
i am going to cherry pick my comments from this PR.

File: dart_sdk/test/pylons_wallet/pylons_wallet_impl_test.dart [hunk 1]
```diff
@@ -1,684 +1,687 @@
-import 'dart:convert';
-
-import 'package:flutter/services.dart';
-import 'package:flutter_test/flutter_test.dart';
```
[the remark this answers is not in the document]
Comment (afti-githobo) [reply dart_sdk/test/pylons_wallet/pylons_wallet_impl_test.dart:4]: cf. https://github.com/Pylons-tech/pylons/issues/1435

File: dart_sdk/lib/src/pylons_wallet/pylons_wallet_impl.dart [hunk 2]
```diff
@@ -218,19 +222,20 @@ class PylonsWalletImpl implements PylonsWallet {
   }
 
   @override
-  Future<SDKIPCResponse<Execution>> txExecuteRecipe(
+  Future<SDKIPCResponse<String>> txExecuteRecipe(
```
[the remark this answers is not in the document]
Comment (afti-githobo) [reply dart_sdk/lib/src/pylons_wallet/pylons_wallet_impl.dart:225]: this and the other related change above can be rolled back - they're from when i was experimenting w/ that call before you fixed it

File: dart_sdk/lib/src/pylons_wallet/pylons_wallet_impl.dart [hunk 3]
```diff
@@ -58,6 +58,9 @@ class PylonsWalletImpl implements PylonsWallet {
 
   Future<SDKIPCResponse> _dispatch<T>(String key, String data,
       {required bool requestResponse}) async {
+    // TODO: there obviously should not be a hard lock here, figure out wtf is actually up
```
[the remark this answers is not in the document]
Comment (afti-githobo) [reply dart_sdk/lib/src/pylons_wallet/pylons_wallet_impl.dart:61]: ipc messages trample each other if sent too quickly in succession. unsure why. hardlock is inelegant but it ducks that issue for a minute.

File: dart_sdk/lib/src/types/trade.dart [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// todo: write this
+
+// import '../generated/pylons/trade.pb.dart' as generated;
```
[the remark this answers is not in the document]
Comment (afti-githobo) [reply dart_sdk/lib/src/types/trade.dart]: yeah, we never had trade handlers here, this is removed

File: dart_sdk/lib/src/types/execution.dart [hunk 5]
```diff
@@ -0,0 +1,104 @@
+// import 'package:fixnum/fixnum.dart';
+//
```
[the remark this answers is not in the document]
Comment (afti-githobo) [reply dart_sdk/lib/src/types/execution.dart]: un-stubbed the execution wrappers

File: dart_sdk/lib/src/pylons_wallet/pylons_wallet_impl.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (afti-githobo) [reply dart_sdk/lib/src/pylons_wallet/pylons_wallet_impl.dart:61]: cf. https://github.com/Pylons-tech/pylons/issues/1433

**Review by kjawadDeveloper2 — APPROVED** [review]

**Comment by afti-githobo** [comment]
tagging in @faisalnaveedRNS @MikeSofaer @Amean for reviews to get this over the line

**Pull request merged**